src/precice/impl/ParticipantState.cpp

```diff
@@ -91,17 +101,17 @@ void ParticipantState::receiveMesh(const mesh::PtrMesh
   checkDuplicatedUse(meshName);
   PRECICE_ASSERT(!fromParticipant.empty());
   PRECICE_ASSERT(safetyFactor >= 0);
-  auto context               = new MeshContext();
```

Comment:
This should ideally not allocate but add this object to the `_receivedMeshContexts`.

As this object contains a shared pointer, you may want to look into move semantics.

If _recievedMeshContexts guarantees persistent element locations in memory (aka no-iterator invalidation), then you can add a pointer to the map.
It is a question of there should be one map per type or one unified map. This may need some thought.